src/net_processing.cpp

```diff
@@ -3744,7 +3744,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             return;
         }
 
-        SetupAddressRelay(pfrom, *peer);
+        Assume(SetupAddressRelay(pfrom, *peer));
```

Comment:
Good idea.

Reply:
The commit log is a good explanation of why you can make this assumption. Perhaps also add something similar as a code comment:

```suggestion
        // SetupAddressRelay will never fail since this must be an outbound connection.
        Assume(SetupAddressRelay(pfrom, *peer));
```

Reply:
>         // SetupAddressRelay will never fail since this must be an outbound connection.

it can only be an *inbound* connection ;)

Reply:
oops!

Reply:
added a comment